### PR TITLE
Adds kyverno namespace to default scrape config

### PR DESCRIPTION
**Pull request by rewanthtammana**
## Description

Adds `kyverno` namespace to default scrape config for WC to look for cadvisor metrics.
Towards https://github.com/giantswarm/giantswarm/issues/24831

## Checklist

I have:

- [x] Described why this change is being introduced
- [ ] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

**Comment by hervenicol** [comment]
You're missing UTs generation for the build to pass: https://github.com/giantswarm/prometheus-meta-operator#how-to-build :wink:

**Review by stone-z — APPROVED** [review]
Cool cool cool

File: files/templates/scrapeconfigs/additional-scrape-configs.template.yaml [hunk 1]
```diff
@@ -96,7 +96,7 @@
     action: keep
```
Comment (stone-z) [inline files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:96]: I know we have service monitors on MCs -- do those also get us the same cadvisor metrics? Maybe we should add this to the MC configuration as well

Comment (hervenicol) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:96]: Yes, I think you should add it to the MC as well.

Comment (rewanthtammana) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:96]: Sure, will do that.

Comment (stone-z) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:96]: @hervenicol does adding the namespace here result in _everything_ in that namespace getting scraped automatically? We have some resources in this namespace in WCs that we don't want to scrape

Comment (hervenicol) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:96]: Only cadvisor data. But all pods in the namespace will be scraped.
If you know how to determine what you don't want to scrape, you could add a `drop` action.

Like:
```
  - source_labels: [container]
    regex: .*IWannaDrop.*
    action: drop
```
Just make sure it does not catch too many metrics!

The current config is a bit hard to read and maintain, hopefully when we switch to serviceMonitors it's easier.

File: files/templates/scrapeconfigs/additional-scrape-configs.template.yaml [hunk 2]
```diff
@@ -92,11 +92,11 @@
   metric_relabel_configs:
 [[- if eq .ClusterType "management_cluster" ]]
   - source_labels: [namespace]
-    regex: (kube-system|giantswarm.*|.*-prometheus|monitoring|draughtsman|flux-.*)
+    regex: (kube-system|giantswarm.*|.*-prometheus|monitoring|draughtsman|flux-.*|kyverno)
```
[the remark this answers is not in the document]
Comment (stone-z) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:95]: ^ sorry, previous comment applies to the WC config, not the MC

[the remark this answers is not in the document]
Comment (rewanthtammana) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:95]: I tested on one of the WC prometheus (on MC) & it's not showing any policy reporter metrics or results. I believe this config fetches only cadvisor data.

As for kyverno app, we didn't add any labels to policy reporter either. So, policy reporter results shouldn't be scraped.

[the remark this answers is not in the document]
Comment (rewanthtammana) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:95]: Also, the installation of `kyverno-app` in WC via `happa` doesn't expose [policy reporter metrics](https://github.com/giantswarm/kyverno-app/blob/main/helm/kyverno/charts/policy-reporter/values.yaml#L101-L102) by default. That's the reason for no scrapes.

image

Upon, enabling `metrics` for `policy-reporter`,

```bash
policy-reporter:
  metrics:
    enabled: true
```

image

I think we can have this disabled to ensure policy reporter pods aren't being scraped on WC. It is enabled for MC.

[the remark this answers is not in the document]
Comment (hervenicol) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:95]: > I believe this config fetches only cadvisor data.

True. It's cadvisor config. So if you want cadvisor data for all containers in the kyverno namespace, that's good :+1:

[the remark this answers is not in the document]
Comment (rewanthtammana) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:95]: Yes, that's exactly what we are looking for 💪

**Review by hervenicol — APPROVED** [review]
LGTM

**Pull request merged**